### PR TITLE
[TIMOB-18351] Deprecate iOS and Android modules in node-appc

### PR DESCRIPTION
**Pull request by skypanther**
https://jira.appcelerator.org/browse/TIMOB-18351

### CODE REVIEW

File: lib/commands/sdk.js [hunk 1]
```diff
@@ -446,6 +446,11 @@ SdkSubcommands.select = {
 					logger.error(__('Invalid Titanium SDK "%s"', value));
 					return false;
 				}
+				if(parseInt(value.split('.').slice(0,3).join('')) < 320) {
```
Comment (cb1kenobi) [inline lib/commands/sdk.js]: You should use node-appc to do your version comparison:

```javascript
if (appc.version.lt(value, '3.2.0')) {
```

Also, make sure you have a space between `if` and the `(`. `if` is not a function.

File: lib/commands/sdk.js [hunk 2]
```diff
@@ -446,6 +446,11 @@ SdkSubcommands.select = {
...
+					console.log('');
```
Comment (cb1kenobi) [inline lib/commands/sdk.js]: You should never use `console.log()`. Use `logger.log()`.

File: lib/commands/sdk.js [hunk 3]
```diff
@@ -446,6 +446,11 @@ SdkSubcommands.select = {
...
+					logger.warn(__('Selected SDK contains deprecated environment detection code.' +
+						'\n        Please choose a 3.2 or newer SDK for more accurate information.'));
```
Comment (cb1kenobi) [inline lib/commands/sdk.js]: Please split this up into 2 separate `logger.warn()` calls. When we translate strings, we do it one sentence at a time. This means you no longer need the `\n` and all the whitespace before "Please".

File: lib/commands/sdk.js [hunk 4]
```diff
@@ -446,6 +446,11 @@ SdkSubcommands.select = {
 					logger.error(__('Invalid Titanium SDK "%s"', value));
 					return false;
 				}
+				if (appc.version.lt(value, '3.2.0')) {
+					logger.log('');
+					logger.warn(__('Selected SDK contains deprecated environment detection code.'));
+					logger.warn(__('Please choose a 3.2 or newer SDK for more accurate information.'));
```
Comment (cb1kenobi) [inline lib/commands/sdk.js]: The warning here needs to change. We shouldn't concern the user with detection code and such. We simply need to say:

```
Titanium SDK {VERSION} has been deprecated and will not work with future releases.
Please use Titanium SDK 3.2 or newer.
```

**Comment by cb1kenobi** [comment]
This commit also needs to include additional logic in the https://github.com/appcelerator/titanium/blob/master/hooks/tisdk3fixes.js file. In the `cli:post-validate` hook, you need to detect if the tiapp.xml version is less than 3.2 and print a big warning saying they need to upgrade the app along with a similar message to the `ti sdk select` warning.

**Comment by pinnamur** [comment]
CR **passed**.

**Pull request merged**